### PR TITLE
AREA-48 - Authentication repository

### PR DESCRIPTION
**Pull request by Yann-Masson**
I added the authentication repository. This repository is only mapping api endPoint with DTOs.
Also, I updated the API `call` function because a function like this one can't really produce a dynamically type so I removed it. And it's the repository who's just mapping request result to a DTO.

### CODE REVIEW

File: mobile/lib/repositories/dtos/authentication.dtos.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  @override
+  Map<String, dynamic> toJson() => _$InLoginDTOToJson(this);
+}
+
+@JsonSerializable()
+class OutLoginDTO implements Json {
+  final String token;
+  final String refreshToken;
+
+  OutLoginDTO({required this.token, required this.refreshToken});
```
Comment (TekMath) [inline mobile/lib/repositories/dtos/authentication.dtos.dart:26]:
```suggestion
  OutLoginDTO({required this.accessToken, required this.refreshToken});
```

File: mobile/lib/repositories/authentication.repository.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return Response<OutLoginDTO>(
+      statusCode: response.statusCode,
+      message: response.message,
+      data: response.data != null ? OutLoginDTO.fromJson(response.data) : null,
+      errors: response.errors,
+    );
+  }
+
+  Future<Response<OutRefreshTokenDTO>> refreshToken(String refreshToken) async {
+    final response = await call<InRefreshTokenDTO>(
```
Comment (TekMath) [inline mobile/lib/repositories/authentication.repository.dart:28]: Refresh token is a GET method that take the token in the header as refreshToken

**Review by TekMath — CHANGES_REQUESTED** [review]

File: mobile/lib/repositories/authentication.repository.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      errors: response.errors,
+    );
+  }
+
+  Future<Response<OutRegisterDTO>> register(
+      String email, String password, String name) async {
+    final response = await call<InRegisterDTO>(
+      method: 'POST',
+      endpoint: '/auth/register',
+      body: InRegisterDTO(email: email, password: password, name: name),
```
[the remark this answers is not in the document]
Comment (Yann-Masson) [reply mobile/lib/repositories/authentication.repository.dart:50]: Yep, ur right. I haven't read this doc ^^ I'll do it

File: mobile/lib/repositories/authentication.repository.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      data: response.data != null
+          ? OutRefreshTokenDTO.fromJson(response.data)
+          : null,
+      errors: response.errors,
+    );
+  }
+
+  Future<Response<OutRegisterDTO>> register(
+      String email, String password, String name) async {
+    final response = await call<InRegisterDTO>(
```
[the remark this answers is not in the document]
Comment (Yann-Masson) [reply mobile/lib/repositories/authentication.repository.dart:47]: Okay my bad, I'll do that

**Review by Thomaltarix — APPROVED** [review]

**Review by TekMath — APPROVED** [review]

**Pull request merged**